edgelet/iotedge/src/check/checks/well_formed_connection_string.rs

```diff
@@ -49,10 +49,21 @@ impl WellFormedConnectionString {
             check.iothub_hostname = Some(hub);
             self.iothub_hostname = check.iothub_hostname.clone();
         } else if check.iothub_hostname.is_none() {
-            let warning = "Device not configured with manual provisioning, in this configuration 'iotedge check' is not able to discover the device's backing IoT Hub.\n\
-                            To run connectivity checks in this configuration please specify the backing IoT Hub name using --iothub-hostname switch if you have that information.\n\
-                            If no hostname is provided, all hub connectivity tests will be skipped.";
-            return Ok(CheckResult::Warning(Context::new(warning).into()));
+            let provisioning_file_path = settings
+                .homedir()
+                .join("cache")
+                .join("provisioning_backup.json");
+
+            let provision_result = provisioning::restore(provisioning_file_path).map_err(|e| {
+                let reason = format!("Could not retrieve iothub_hostname from provisioning file.\n\
+                Please specify the backing IoT Hub name using --iothub-hostname switch if you have that information.\n\
+                If no hostname is provided, all hub connectivity tests will be skipped.\n\
+                Provisioning file error: {}", e);
```

Comment:
Remove this last line. You're already using this message as a context of the original error, so it'll already get printed. There's no reason to print it twice.